bots/csr/src/main/java/org/openjdk/skara/bots/csr/PullRequestWorkItem.java

```diff
@@ -248,14 +253,30 @@ public Collection<WorkItem> run(Path scratchPath) {
         if (needToAddUpdateMarker) {
             addUpdateMarker(pr);
         }
-        if (allCSRApproved && existingCSR && pr.labelNames().contains(CSR_LABEL)) {
+        if (allCSRApproved && existingCSR && !withdrawnCSRWhenCSRNeeded && pr.labelNames().contains(CSR_LABEL)) {
             log.info("All CSR issues closed and approved for " + describe(pr) + ", removing CSR label");
             pr.removeLabel(CSR_LABEL);
         }
         logLatency();
         return List.of();
     }
 
+    /**
+     * Determine whether the CSR label is added via '/csr needed' command
+     */
+    private boolean CSRNeeded(List<Comment> comments) {
+        var CSRNeededCommentCount = 0;
+        for (var comment : comments) {
+            if (comment.body().contains("<!-- csr: 'needed' -->")) {
+                CSRNeededCommentCount++;
+            }
+            if (comment.body().contains("<!-- csr: 'unneeded' -->") && CSRNeededCommentCount > 0) {
+                CSRNeededCommentCount--;
+            }
+        }
+        return CSRNeededCommentCount > 0;
```

Comment:
I don't think this should be determined by count. The last comment with `<!-- csr: '[un]?needed' -->` in it is the one that is currently in effect.

Reply:
Agree with you, I will change it.